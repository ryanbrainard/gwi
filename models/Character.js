import VoicePlayerMRU from "./VoicePlayerMRU";

export default class Character {
  static _chars = {};

  static register(char) {
    this._chars[char.name] = char;
    return char;
  }

  static find(name) {
    return this._chars[name];
  }

  constructor(name, group, voices) {
    this._name = name;
    this._group = group;
    this._voices = voices;

    this.player = new VoicePlayerMRU(this._voices);
  }

  get key() {
    return this.name;
  }

  get name() {
    return this._name;
  }

  get group() {
    return this._group.map(c => Character.find(c));
  }

  async preloadPlay() {
    await this.player.load();
  }

  async play() {
    await this.player.play();
  }
}
