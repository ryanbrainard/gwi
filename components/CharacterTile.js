import PropTypes from "prop-types";
import React from "react";
import {
  Dimensions,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from "react-native";
import config from "../config";
import Character from "../models/Character";

export default class CharacterTile extends React.Component {
  static propTypes = {
    char: PropTypes.instanceOf(Character).isRequired,
    size: PropTypes.number,
    color: PropTypes.string,
    show: PropTypes.bool
  };

  static defaultProps = {
    show: true,
    percentOfWidth: 0.3
  };

  render() {
    const { char, size, color, show } = this.props;

    return (
      <View style={styles.container}>
        <TouchableOpacity
          style={[
            styles.button,
            {
              backgroundColor: color,
              width: size,
              height: size
            }
          ]}
          onPress={char.play.bind(char)}
        >
          {<Text style={styles.text}>{show ? char.name : "?"}</Text>}
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    borderWidth: 2,
    borderColor: config.colors.background
  },
  button: {
    alignItems: "center",
    justifyContent: "center"
  },
  text: {
    fontWeight: "bold",
    fontSize: 24,
    color: config.colors.text
  }
});
