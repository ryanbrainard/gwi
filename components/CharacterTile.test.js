import React from "react";
import renderer from "react-test-renderer";
import Character from "../models/Character";
import CharacterTile from "./CharacterTile";

test("renders correctly", () => {
  const char = new Character(); // TODO: fill in

  const tree = renderer.create(<CharacterTile char={char} />).toJSON();
  expect(tree).toMatchSnapshot();
});
