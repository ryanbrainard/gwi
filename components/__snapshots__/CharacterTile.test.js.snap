// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "white",
      "borderWidth": 2,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "height": 225,
        "justifyContent": "center",
        "opacity": 1,
        "width": 225,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    />
  </View>
</View>
`;
